processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -181,6 +189,11 @@ public void validateTwoSegments(final IndexableAdapter adapter1, final Indexable
 
   public QueryableIndex loadIndex(File inDir) throws IOException
   {
+    return getSegmentizerFactory(inDir).loadIndex(inDir);
+  }
+
+  public QueryableIndex loadIndexDirectly(File inDir) throws IOException
```

Comment:
Readers and users will never know the difference between loadIndex() and loadIndexDirectly(). Maybe move loadIndexDirectly() to MMappedQueryableSegmentizerFactory. Along with some other methods.